File: 0713-subarray-product-less-than-k/0713-subarray-product-less-than-k.py
class Solution:
    def numSubarrayProductLessThanK(self, nums: List[int], k: int) -> int:
        if k<=1:
            return 0
        left =0
        c=0
        p=1
        for right in range(len(nums)):
            p=p*nums[right]
            while p>=k:
                p=int(p/nums[left])
                left+=1
            c+=right-left+1
        return c

File: 0713-subarray-product-less-than-k/0713-subarray-product-less-than-k.cpp
class Solution {
public:
    int numSubarrayProductLessThanK(vector<int>& nums, int k) {
        if (k<=1) return 0;
        int n=nums.size();
        int p=1;
        int left=0;
        int c=0;
        for( int right=0;right<n;right++){
            p*=nums[right];
            while(p>=k){
                p/=nums[left];
                left++;
            }
            c+=right-left+1;
        }
        return c;
    }
};